iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/HardCodedCredentialsCheck.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.checks;
+
+import java.util.List;
+import java.util.function.Predicate;
+import org.sonar.check.Rule;
+import org.sonar.iac.arm.checkdsl.ContextualResource;
+import org.sonar.iac.arm.tree.api.ArmTree;
+import org.sonar.iac.arm.tree.api.Expression;
+import org.sonar.iac.arm.tree.api.ResourceDeclaration;
+import org.sonar.iac.arm.tree.api.StringLiteral;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+
+@Rule(key = "S6437")
+public class HardCodedCredentialsCheck implements IacCheck {
+
+  private static final String MESSAGE = "Revoke and change this secret, as it might be compromised.";
+
+  private static final List<String> CREDENTIAL_PROPERTIES = List.of(
```

Comment:
The list can be extended by `publishingPassword`, `publishingUserName`, and probably more names.
Example: 
```
{
  "type": "Microsoft.Web/publishingUsers",
  "apiVersion": "2022-09-01",
  "name": "web",
  "kind": "string",
  "properties": {
    "publishingPassword": "string",
    "publishingPasswordHash": "string",
    "publishingPasswordHashSalt": "string",
    "publishingUserName": "string",
    "scmUri": "string"
  }
}
```

From: https://learn.microsoft.com/en-us/azure/templates/microsoft.web/publishingusers?pivots=deployment-language-arm-template

Have you tried to find more credential properties?